mirage/views/private/utils.ts

```diff
@@ -0,0 +1,259 @@
+        sortKey = queryParams.sort;
+    }
+    return data.sort(dynamicSort(sortKey));
+};
+
+export const buildQueryParams = (params: QueryParameters): string => {
+    let paramString = '?';
+    Object.keys(params).forEach(key => {
+        if (paramString.length > 1) {
+            paramString = `${paramString}&`;
+        }
+        let encodedValue = params[key];
+        if (encodedValue !== null) {
+            encodedValue = encodeURIComponent(encodedValue);
+        }
+        paramString = `${paramString}${key}=${encodedValue}`;
+    });
+    if (paramString.length > 1) {
+        return paramString;
+    } else {
+        return '';
+    }
+};
+
+export const paginate = (request: any, data: any[], options: ProcessOptions): JsonData => {
+    const total = data.length;
+    const { queryParams, url } = request;
+    const self = `${url}${buildQueryParams(queryParams)}`;
+    let start: number = 0;
+    let { defaultPageSize } = options;
+    if (defaultPageSize === undefined) {
+        defaultPageSize = 10;
+    }
+    let perPage = defaultPageSize;
+    let currentPage = 1;
+    if (typeof queryParams === 'object') {
+        if ('page[size]' in queryParams && queryParams['page[size]'] !== 0) {
+            perPage = queryParams['page[size]'];
+        }
+        if ('page' in queryParams) {
+            currentPage = queryParams.page;
+            start = (currentPage - 1) * perPage;
+        }
+    }
+    const pages = Math.ceil(total / perPage);
+    const nextPage = Math.min(currentPage + 1, pages);
+    const prevPage = Math.max(currentPage - 1, 1);
+    const lastPage = pages;
+
+    let prev = null;
+    let next = null;
+
+    queryParams.page = 1;
+    const first = `${url}${buildQueryParams(queryParams)}`;
+    queryParams.page = lastPage;
+    const last = `${url}${buildQueryParams(queryParams)}`;
+    if (nextPage > currentPage) {
+        queryParams.page = nextPage;
+        next = `${url}${buildQueryParams(queryParams)}`;
+    }
+    if (prevPage < currentPage) {
+        queryParams.page = prevPage;
+        prev = `${url}${buildQueryParams(queryParams)}`;
+    }
+
+    const paginatedJson = {
+        data: data.slice(start, start + perPage),
+        links: {
+            self,
+            first,
+            next,
+            prev,
+            last,
+        },
+        meta: {
+            total,
+            per_page: perPage,
+        },
+    };
+    return paginatedJson;
+};
+
+export const embed = (schema: any, request: any, json: JsonData, config: any) => {
+    const { queryParams } = request;
+    const { data } = json;
+    if ('embed' in queryParams && queryParams.embed !== {}) { // If we have embeds query params
+        let embedKeys = [];
+        if (Array.isArray(queryParams.embed)) {
+            embedKeys = queryParams.embed;
+        } else {
+            embedKeys.push(queryParams.embed);
+        }
+        for (const embedded of embedKeys) { // Go through the embed keys
```

Comment:
Add the alwaysEmbed keys to embedKeys before here to ensure we get the auto-embeds even when grabbing an item directly.